cnvlib/segmentation/__init__.py

```diff
@@ -12,11 +12,42 @@
 from ..cnary import CopyNumArray as CNA
 from . import cbs, flasso, haar
 
-from Bio._py3k import StringIO
+from concurrent import futures
 
+from Bio._py3k import StringIO
 
 def do_segmentation(cnarr, method, threshold=None, variants=None,
                     skip_low=False, skip_outliers=10,
+                    save_dataframe=False, rlibpath=None,
+                    processes=1):
+    """Infer copy number segments from the given coverage table."""
+    if processes == 1:
+        return _do_segmentation(cnarr, method, threshold, variants,
+                                skip_low, skip_outliers,
+                                save_dataframe, rlibpath)
+
+    chroms = cnarr.chromosome.unique()
+    rets = []
+    # TODO: handle save_dataframe=True
+    with futures.ProcessPoolExecutor(processes) as pool:
+        for ret in pool.map(_ds, ((cnarr[cnarr.chromosome == c], method,
+                                   threshold, variants, skip_low, skip_outliers,
+                                   False, rlibpath) for c in chroms)):
```

Comment:
You can do:

    ((chromarr, method, ...) for _c, chromarr in cnarr.by_chromosomes())

and avoid creating `chroms` above. (Not that this is a bottleneck, but it's cleaner.)